lib/core/jmanagedbackends.c

```diff
@@ -0,0 +1,832 @@
+/**
+ * \file JManagedBackends.c
+ * \ingroup backend
+ *
+ * \copyright
+ * JULEA - Flexible storage framework
+ * Copyright (C) 2022 Julian Benda
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+#include <bits/types/struct_timeval.h>
+#include <core/jmanagedbackends.h>
+#include <core/jbackend.h>
+#include <core/jlist.h>
+#include <core/jlist-iterator.h>
+#include <core/jsemantics.h>
+#include <core/jtrace.h>
+#include <core/jconfiguration.h>
+
+#include <julea-config.h>
+
+#include <glib.h>
+
+#include <string.h>
+#include <sys/select.h>
+#include <sys/time.h>
+
+#define EXE(cmd, ...) \
+	do \
+	{ \
+		if ((cmd) == FALSE) \
+		{ \
+			g_warning("EXE failed at %s:%d with:", __FILE__, __LINE__); \
+			g_warning(__VA_ARGS__); \
+			goto end; \
+		} \
+	} while (FALSE)
+
+/// an read write spin lock to avoid data access on migration
+typedef struct
+{
+	gint read_access_count; ///< number of read access
+	guint generation; ///< generation count, incremented after each write_access
+	gint write_access; ///< true if a write operation is scheduled or running
+} RWSpinLock;
+
+/// acquired a read lock
```

Comment:
Please check doxygen comment style. (Also for following functions in this file.)